packages/app/src/components/Root/Root.tsx

```diff
@@ -49,13 +64,124 @@ const SideBarItemWrapper = (props: SidebarItemProps) => {
 
 export const MenuIcon = ({ icon }: { icon: string }) => {
   const app = useApp();
+  const SystemIcon = app.getSystemIcon(icon);
+
+  if (SystemIcon) {
+    return <SystemIcon fontSize="small" />;
+  }
+
+  if (icon.startsWith('<svg')) {
+    const svgDataUri = `data:image/svg+xml;base64,${btoa(icon)}`;
+    return (
+      <MuiIcon style={{ fontSize: 20 }}>
+        <img src={svgDataUri} alt="" />
+      </MuiIcon>
+    );
+  }
+
+  if (icon.startsWith('http')) {
+    return (
+      <MuiIcon style={{ fontSize: 20 }} baseClassName="material-icons-outlined">
+        <img src={icon} alt="" />
+      </MuiIcon>
+    );
+  }
+
+  return (
+    <MuiIcon style={{ fontSize: 20 }} baseClassName="material-icons-outlined">
+      {icon}
+    </MuiIcon>
+  );
+};
+
+const renderIcon = (iconName: string) => () => <MenuIcon icon={iconName} />;
 
-  const Icon = app.getSystemIcon(icon) || (() => null);
-  return <Icon />;
+const renderExpandIcon = (expand: boolean) => {
+  return expand ? (
+    <ExpandMore
+      fontSize="small"
+      style={{ display: 'flex', marginLeft: '0.5rem' }}
+    />
+  ) : (
+    <ChevronRightIcon
+      fontSize="small"
+      style={{ display: 'flex', marginLeft: '0.5rem' }}
+    />
+  );
 };
 
 export const Root = ({ children }: PropsWithChildren<{}>) => {
-  const { dynamicRoutes, mountPoints } = useContext(DynamicRootContext);
+  const { dynamicRoutes, menuItems } = useContext(DynamicRootContext);
+  const [openItems, setOpenItems] = useState<{ [key: string]: boolean }>({});
+
+  const handleClick = (itemName: string) => {
+    setOpenItems(prevOpenItems => ({
+      ...prevOpenItems,
+      [itemName]: !prevOpenItems[itemName],
+    }));
+  };
+  const renderExpandableMenuItems = (
+    genericMenuItem: ResolvedGenericMenuItem,
+    isOpen: boolean,
+  ) => {
+    return (
+      <>
+        {genericMenuItem.children.length > 0 &&
+          genericMenuItem.children?.map(child => (
+            <Collapse key={child.name} in={isOpen} timeout="auto" unmountOnExit>
+              <List component="div" disablePadding>
+                <ListItem component="div" disableGutters>
+                  <SidebarItem
+                    icon={renderIcon(child.icon)}
+                    text={child.title}
+                    to={child.to ?? ''}
+                    style={{ paddingLeft: '2rem' }}
+                  />
+                </ListItem>
+              </List>
+            </Collapse>
+          ))}
+      </>
+    );
+  };
+
+  const renderGenericMenuItems = (isBottomMenuSection: boolean) => {
+    const menuItemArray = isBottomMenuSection
+      ? menuItems.filter(mi => mi.name === 'admin')
+      : menuItems.filter(mi => mi.name !== 'admin');
+    const sortedMenuItemArray = [...menuItemArray].sort(compareMenuItems);
```

Comment:
Please sort the elements in the provider (in the extract function). That makes it also easier to write unit tests for the different priorities.